### PR TITLE
Adds writing of wifi-credentials and cleans up some code

### PR DESCRIPTION
**Pull request by tommyknows**
This MR implements the writing of WiFi credentials to `auth.h`.
Also, it implements the `cleanup` routine that can be used to remove all images.
In addition, it adds the `trap`, to do cleanup work on failure / exit of the script. This allows for better debug messages.

### CODE REVIEW

File: Device/build.sh [hunk 1]
```diff
@@ -115,7 +155,25 @@ stop_build_container() {
     docker rm "$SETUP_CONTAINER_ID"
 }
 
+write_wifi_credentials() {
```
Comment (anliksim) [inline Device/build.sh:158]: Why don't you use 'sed' to just replace the user and password?

Comment (tommyknows) [reply Device/build.sh:158]: I guess it's just a matter of preference. If the file doesn't exist, we're safe with that too.

But it got me thinking, what if the user does not want to specify WiFi credentials and wants to login manually?
Is it enough to just reset the chip again? 
If not, we would need to remove "all" the code that contains the wifi setup. (?)

**Comment by tommyknows** [comment]
@anliksim can we merge this and create a fix for the "no wifi-credentials" problem later?

**Comment by tommyknows** [comment]
closing this as i will be opening more complete MR.

**Pull request closed without merging**